drv/transceivers-server/src/udp.rs

```diff
@@ -529,48 +530,34 @@ impl ServerImpl {
         }
     }
 
-    /// This should be called as the last operation before sending the response
-    /// since error details is the last of the trailing data. This function
-    ///
-    /// Panics
-    ///
-    /// This function panics if `None` is the supplied `failure_type` and the
-    /// `result` contains any  bits set in the failure mask as the programmer
-    /// must supply a `HwError` when calling this function on a `result` which
-    /// may have a non-zero failure mask.
-    fn handle_errors(
+    /// This function reads a `ModuleResult` and populates and failure or error
+    /// information at the end of the trailing data buffer. This means it should
+    /// be called as the last operation before sending the response. For results
+    /// where a `ModuleResultNoFailure` is returned, use handle_errors instead.
+    fn handle_errors_and_failures(
```

Comment:
Optional API suggestion: this function (and `handle_errors`) could be a little cleaner if they accepted a pre-sliced `out`, instead of also taking `data_idx`.  This would make it the caller's responsibility to slice the buffer, but the most common caller is passing in `0`, which wouldn't require any slicing!

The only place you'd have to be careful is in cases like
```rust
let (final_payload_len, errored_modules) =
    self.handle_errors(modules, result, num_status_bytes, out);
```
which would become
```rust
let (err_len, errored_modules) =
    self.handle_errors(modules, result, &mut out[num_status_bytes..]);
let final_payload_len = num_status_bytes + err_len;
```

Reply:
It is definitely cleaner and less complicated. Done.